### DISCUSSION TITLE
Issues with Tasmota Hue emulator or with Alexa?  Either way help!

### DISCUSSION THREAD

**Discussion by CancunManny** [arendst/Tasmota#22227] [Q&A] [answered]
Long time user of Tasmota, and NEVER had this issue before.  I have 12 tasmota entities with about 23 devices that Alexa could usually find. Today that I paid attention seems the version of tasmota ranges from version 13 to 14. (will update all of them to newest version soon, and they are all esp8266s d1 minis except for two smart bulbs that I flashed with tasmota).

In one of the tasmotas I was using one of the pins that are used at bootup.  During power outages sometimes it would bootup, other times I would have to open up the switch box and thinker with it.  Today I had to cut the power off for a bit, and when I turned power back on it wouldn't boot.  Figured today was the day to re configure the tasmota so I wouldn't have that issue anymore.  Not sure what I did, but seems I ended up burning the esp8266 (d1 mini).  I have a few extras laying around, and through experience I did happen to have a backup config file.

I flashed a new d1 mini, loaded the config file I had for it, and figured I was good to go.  This device has 3 relays (Sol, Cueva, Viento), 1 PMW (Sol Rojo), 1 button and 1 PIR switch.  The button worked as expected using multi tap to control the different relays.  The switch is detached.  All my D1 Minis were set up as Hue Emulator multi device.

I then tried to control it via Alexa and it wouldn't work.  I found it odd, but figured maybe the IP of the device was the issue, since this new D1 Mini was getting a different IP.  At that point I decided to do what I usually do, got my phone out, opened up the Alexa app, and deleted Sol, Cueva, Viento and Sol Rojo, then asked Alexa to "find new devices".

After the search, Alexa said it found ONE new device which was odd since it should have found 4.  All other tasmotas/d1 minis were working fine with Hue Emulator turned on.  I did a quick google search, and found someone had similar issue back in 2022 and they were able to solve it by going to Amazon's page, Alexa privacy settings, then deleting all device history.  

Don't ask me why, but before deleting all device history I decided to delete ALL entities that Alexa had found (and that were working properly).  Then I deleted all device History from Alexa, and asked to find new devices.

It said it found 20 devices, but when I started testing them with Alexa most wouldn't work.  That is when I noticed something extremely odd happening.  Using the D1 Mini I replaced as an example.  On Alexa app it would show only one entry (usually viento), I would click on it, it would tell me not responding, I would select delete, and then it would take me back to the device list.  Well on the device list the same name would pop up again (lets say viento), when I clicked on viento, it would show another name (like Sol), so I deleted that one also.  It would then take me back to devices list, and "viento" would pop up again, clicked on it, and on the new screen it would show another entity on the device like Cueva.  Basically I had to delete the device 4 times.

Basically all my d1 minis that had more than one relay/pmw on it I had to delete many times.  A couple times it was a bit different.  For example at one point the name listed on devices was Sol.  When I clicked to open sol menu to delete it actually showed "sol".  When I deleted that and went back to the main device list "viento" popped up which wasn't there before I deleted "sol".

It was time to do extended troubleshooting.  First thing I decided to do was to log into each of my tasmota devices and turn off the emulators.  Once all emulators were set to "none", I deleted all the devices from the Alexa list, and asked Alexa to find new devices.  I was assuming it wasn't going to find any new devices.  I was wrong, it found the 20 devices again!  At this point I start scratching my head as I am pretty sure Hue Emulator doesn't use mqtt so it wouldn't be an mqtt retain issue.

On my HA I do have the Alexa hacs installed, and it does share devices attached to Alexa, but since I don't have Nabu Casa my HA devices are not exposed to Alexa.  I read there is a way to run the HUE emulator on HomeAssistant in order to expose scripts/automations to Alexa, I was pretty sure I never did set that up, but just to be on the safe side I shut down my HA server.

So with HA server shutdown, all tasmota devices set to emulator none, I went ahead and deleted all devices from Alexa app, then went in and deleted all device history from Alexa again.  I then asked Alexa to find new devices, and AGAIN it found 20 of them!  I still have NO IDEA where it was getting that info from,

I then deleted Sol, Cueva, Viento and Sol Rojo again from the Alexa app (left the rest untoched).  I decided to log in to the tasmota configuration and selected the wemo single device emulator.  When I played with this in the past it would only pick up the first friendly name (first device) and ignored the rest.  To my surprise when I asked Alexa to find new devices it found 4 new devices.  All 4 of them show as switches/plugs but I can control all 4 of them via voice.

I then did the same for a second room, except this time I forgot to delete the device it had found first.  This room only has to relays, and to my surprise it found them as wemo plugs, and voice control with Alexa was working.  On the Alexa device list I could see the two new plugs (Vale Luz and Vale Ventilador), and was also showing Vale Luz as a light bulb with the "not responding" message attached to it.

I once again got into the tasmota configuration for both of the ones I had working a wemo, and set emulation to none.  All my tasmotas were setup to no emulation.  I went ahead and deleted all devices from Alexa list again.  I then asked Alexa to find new devices, and this time it found no new devices.  That made sense to me!

Out of my 12 tasmotas, 2 are smart blubs and I set those up as Hue emulators.  The other 10 I set up as wemo emulator.  I asked Alexa to find new devices and it found all my devices.  As far as voice control everything is back to how I had it before, except most lights now show as plugs on the Alexa app.  The only draw back is 2 or 3 led lights I have set up as PWM.  Before I could adjust intensity by voice, but now I can only turn on/off.

I am pretty sure the Hue emulator itself didn't stop working for everyone, as I am sure i would have found many recent topics about it here in the tasmota Q & A section or in the issues section.  That said any ideas of what I could try to get Hue Emulator working again for me?  Something tells me it has something to do of where Alexa was finding the 20 devices even though all emulators were turned off.

*Edit*  So I keep doing some research, and seems echo devices being used is important.  I have 6 echo 3rd generation, and one echo 4th generation.  Most of the time the "find new devices" was done at the echo 4th gen. When I set up the second room with Wemo it was done with an echo 3.

**Comment by s-hadinger** [comment]
Sometimes you need to trigger multiple discovery from Alexa. It looks like some echo devices keep a memory or queue of previously discovered devices. After some time it stabilizes again

**Reply by CancunManny** [reply]
Did some extra reading last night before going to bed.  Some say that doing factory reset in all my echos might do the trick.  Someone shared a log and it was "one" offending device throwing the hue emulator off.  Will try that and report my findings.

**Reply by s-hadinger** [reply]
Factory setting your echo devices looks to me as an overkill. But if you do it, please report if it helped

**Reply by CancunManny** [reply]
So first attempt failed.  I unplugged all echo devices.  Turned off all emulators in tasmotas. Deleted all the "plugs" I had on the Alexa app.  Deleted Smart Home Device history from Amazon.com (not Alexa.amazxon.com).  Put my main echo device (it is an Echo 5th gen) in pairing mode, and connected to it.  Asked it find new devices, and it found all 23 plugs again!!!  Have no idea where it stores the info.

Going by memory but I don't think I had found new devices with the echo 5 before, by the time I got it all was already set up with the older echos.  Next attempt will be to try the same, but this time use one of the gen 3 echo to do the search, while all other echos are powered off.

**Reply by CancunManny** [reply]
This is insane!  So I deregistered ALL of my echos, they are all unplugged.  I confirmed all emulators are off.  I then set up an echo dot 3rd gen, and again it finds all the plugs even though they are not supposed to be found!

Next step is to call Alexa customer service to see if they can help me out as the issue seems to be with them

**Reply by CancunManny** [reply]
Seems Alexa Hue Emulator is no longer working. I will open an "issue" on the issues section.  

I spent over an hour with Amazon and they couldn't help.  I was assuming there was an issue with my actual account that was preventing the lights be found by Alexa.  I disconencted and deregistered ALL my echo devices.  Then I did a factory reset on of them, a 3rd gen one.  

I then created a NEW amazon account.  I logged in with the new amazon account to my Alexa app.  With the new alexa app I connected to the echo I did the factory reset on, and brought it online.  I asked Alexa to find new devices and it didn't find any, as expected.  I then turned one of the hue emulators on via the tasmota configugration, asked Alexa to find new devices.  

Alexa said it found one device named Cueva.  (It should have found 4 devices, Sol, Cueva, Viento and Sol Rojo).  On the Alexa app device list is showed "Sol" as a light bulb.  When I clicked on it, it read "viento".  So as before I ended up deleting 4 devices, but only one of them would show up at a time on the device list. 

That pretty much tells me the issue IS NOT with something "bad" stored on my Alexa app, which means it has to be an issue with the Hue emulator.

Thankfully the BilKin WeMo emulator now can find more than "one" device, otherwise I would have pretty much lost most of my Alexa voice controls, which is what my family uses the most.

I like the idea of having voice assistant work locally.  I actually configured a couple of devices.  Turns out  in English it works good, not as good as Alexa but definitely usable.  However when I tried to set it up to work in spanish it barely works.

**Reply by CancunManny** [reply]
So I am back to square one.  The issue is IN my setup, that is why even with a new Amazon account I end up with the problem.  

After deleting everything from my Alexa app, re configuring the echos etc I could only get it to work with Wemos Emulator.  I decided to do one more test.  I actually opened up the switch box and pulled out the 5v pin of the tasmota that started my nighmare.  All my other tasmotas were setup to wemos emulator.  I then logged in to one of them, and set it up as a Hue emulator.  When I asked Alexa to find new devices it found ALL 24 of them, even the 4 of the tasmota that is powered down, and the rest of them that had the wemos emulator.

At that point I figured I was on the clear, that the new tasmota I set up last night was the one causing all the issues.  Since I now had all my devices listed twice on the alexa app (once as wemos plugs and onces a philips) I decided to start fresh and delete ALL the devices again.....  I then switched all tasmotas to hue emulator, and now I am back to square one where Alexa doesn't find my devices correctly.

Where is Alexa pulling the devices from?  Is there something I can clear on my network?  What is trowing me off is that it did the same thing with a fresh account of Amazon, where it wouldn't find new devices correctly.

**Comment by Jason2866** [comment]
Have you tried to switch off the device(s) where the Alexa app is installed?
If not you can try this. In my experience the app is disturbing the search of the Tasmota HUE emulator devices.

**Reply by CancunManny** [reply]
At this point I've tried pretty much everything, except turning off my phone (the only place I have the alexa app installed).  Will give that a try tomorrow and report back.

**Comment by spechto** [comment]
Hello, I've the same problem with Alexas device detection. Instead of 4 Lamps Alexa found only 1. When deleting this 1 lamp, a new lamp of the remaining 3 appears and so on, until all 4 Lamps are deleted. It looks like that Alexa found all 4 lamps but put them together on one switch with different access names. There should be 4 Lamps with 4 switches in the Alexa app. Have you got any further? 
In the Weblog of the Tasmota Console I can see that Tasmota reports all 4 devices back to Alexa. I believe that the problem lies with Alexa and not with Tasmota. The question would then be, how can we look behind the scenes at Alexa, i.e. are there debugging options here?

**Reply by CancunManny** [reply]
OMG so finally someone reports the same issue as me!  I wasn't able to figure it out.  The work around is to have tasmota emulate Belkin Wemos.  Thankfully belkin wemos now finds more than just one device.  I spent some time with Amazon/Alexa tech support bu they were not much help.  I even created new Alexa account and got same results.  I was contemplating writing an email to Amazon support asking if they could forward the email higher up to someone that really really knows what Alexa is doing in the background.  

My issue started when I flashed a second tasmota with same info the previous one had, not sure if that had anything to do with it.  I assume you had set up some tasmotas the same way before.  Are those still working correctly?  Did you do anything different this time around?   And finally  do you know people that use tasmotas same way we do.  Would be interesting to see if this is an issue that affects everyone trying to discovery new devices, or if they are isolated cases.    I am thinking they are isolated cases otherwise there would be many more reports of people having this issue.

**Comment by spechto** [comment]
Hello, 

I have put a new Tasmota SONOFF mini R4 into operation, which works so far with the HUE emulation, but it only has one relay. The old 4CH SONOFF didn't cause any problems until I had the idea of renaming all the switch names. To do this, I deleted all the switches in Alexa and tried to have them recognised again (HUE emulation). With the known problem that somehow all switches are recognised, but only one control element is assigned to them in Alexa. So, unfortunately, only one of the four lamps can be controlled with different switch names. I previously had older firmware on the SONOFF and went back to older firmware as a test, but I may have to go back to "very old" firmware as I no longer know exactly what version I had before. 

However, I believe that Amazon has changed something in the HUE recognition that is causing this problem. Only multi-channel devices that are addressed directly via Alexa are likely to be affected. Anyone working with the newer HUE hubs (without Tasmota) also needs an additional app, if I'm correctly informed. 

But it's nice that all devices are now recognised via WeMo emulation. It's interesting to note that the current version of the Tasmota firmware also says ‘Single Device’. It may not have been changed here somthing in the code of Tasmota, but possibly in the Alexa recognition routine?

I have sent an email to Amazon support, but we'll see what comes of it. 

Best Regards
Martin

**Reply by CancunManny** [reply]
I will continue to find a solution, and if I find one I will be sure to come back here and post it.

I don't think it is a tasmota release problem.  I have 11 tasmotas, pretty much all of them with 2 relays or more, some wit PMW.  At one point everything was working fine with Hue emulator.  My tasmota versions are different from one to another.  Seems I install the latest version when I first flash it but I don't go back to update them.  All of mine are version 13 or higher.  I get the same issue with ALL of them.

Like I mentioned, I was considering emailing Alexa/Amazon but since I doubt they officially support the Tasmota Hue Emulator not sure how willing they will be into looking at the problem.  I did spoke to about 4 or 5 different level 1 and level 2 Alexa support staff, and they had no idea what I was talking about.

For example, I did many delete device hisotories from the Alexa and Amazon page, since I know it is pulling history from somewhere.  At one point I had all my devices discovered as wemos plugs.  I did something, and all of the sudden ALL my old/deleted hue discoveries were found again.  I basically had all my devices listed twice, once as a plug and once as a Hue.  Figured I had solved my problem, so I deleted everything from the Alexa app again, switched all my tasmotas back to Hue, and asked Alexa to discover, and well went back to the same issue.  So pretty sure Alexa keeps a history of deleted devices somewhere.

Also the point you made about the belkin wemos.  I know in the past if I had 2 or more relays, the wemos emulator would only find the first plug.  Latest tasmota still mentions "single device', but Alexa is finding all of them.  That also leads me to believe it was a change on Alexa side, not on tasmota side. 

What email address did you use to contact Amazon?  I could send an email to the same address and perhaps since there are "two" of us now it might help convince them to forward it to a higher support level.

**Reply by Bigdude65** [reply]
Hi Mate, I have the same problem. Single relay devices work ok. Multi relay devices do not. A Sonoff T2 with 3 relays has 3 friendly names. They are for example X,Y,Z. Alexa will detect X only, but may not necessarily control X. If I delete and redetect, it may control Y or Z. If I delete and redetect, I may get Z but soon changes to X. If I use Belkin WeMo emulator, I only ever get the first relay. 
I have just tried Tasmota 14.2.0 - no change.

**Reply by CancunManny** [reply]
OK, this seems to confirm Alexa/Amazon made changes recently, so I am confident there will be more people popping up with the same issue.

I have the same issue so I totally understand your explanation, but for those not dealing with it, hard to get the point across right?

What I find interesting is that you say if you use Belkim Wemo it only detects the first one.  That is how it used to be, but for me spechto when we asked Alexa to find new devices, it found all of them.

Based on what you wrote, you might be doing things a bit different.  Seems you are deleting one entry and having Alexa find it again.  You have x, y and z.  When you searched alexa said it found "one" device, but it should have found 3.  On you Alexa apps, on device list find the one that it supposedly found.  Click on that one to delete.  On the app list it might say it is Z but when  you click on it so that you can delete it might say it is X.  Regardless of what it says go ahead an delete it.  Once you delete it should take you back to the apps list.  After a second or two the one you tried to delete will pop up again, or one of the ones it should have found will pop up again.

Basically you will have to delete 3 entries, but you won't be able to see the 2nd and 3rd entries  until you delete the 1st one. Once you have deleted all 3 of them, make sure you set up belmin wemos emulator (it does continue to say only for single device).  Then ask Alexa so search for new devices again.  This time it should find all 3 but it finds them as plugs. 

I just tried this 2 days ago and it worked.

**Reply by Bigdude65** [reply]
Thanks for that info, with regard to finding new devices, are you using the "Discover Devices" command, or clicking on the + and going through the manual process?
Cheers

**Reply by Bigdude65** [reply]
Update, - Delete, comes back, delete, comes back, delete comes back, delete ...........didn't come back. Said, Alexa, discover devices and there they are, the 3 friendly names and they work. Thanks for that. great work.
Now to repeat another 28 times...

**Reply by CancunManny** [reply]
Ok, yes I usually discover by asking Alexa to find new devices.  

As far as doing it 28 times.  When I noticed the issue it was tied to "one" device I had to change.  The rest, the ones that had been previously found, were working fine.  I just made the mistake to delete ALL of them figuring I would just re discover everything, as I was getting confused.

What I am getting at, is that as far as I know if I wouldn't have deleted the previous devices it had found, they would have continue to work properly.  The issue only affected new devices.

**Reply by Bigdude65** [reply]
For me, Im moving away from Openhab so they all need to go through the same process.
Cheers

**Comment by s-hadinger** [comment]
It's still unclear what's happening here. I have no explanation, and I can confirm I have multiple Hue emulation lights paired to Alexa and they are discovered and working perfectly well. They are even using Zigbee which adds some complexity for some of them.

**Reply by CancunManny** [reply]
The issue is when you try to discover new devices correct?

**Reply by Bigdude65** [reply]
I also have multiple Tasmota color globes and they work perfectly. I never had any trouble setting them up. Nor did I have any trouble with single relay devices which for me worked seemlessly on Hue or Belkin emulation. The trouble for me was with the multiple relay devices, ie a wall switch with 3 switches and therefor three separate items to discover. This is now fixed Following CancunManny's instructions

**Reply by TheLukeDude** [reply]
I also have issues with devices that used to work perfectly fine with Alexa via Hue emulation. I have several relays with Shellys stock firmware and most relays work with Tasmota. Those with Tasmota are not working reliably (or at all) with Alexa since a few days/maybe two weeks. Hope there's a possibility to fix this in Tasmota.

**Reply by TvTim** [reply]
Hi, I am also having issues, I have several H801 devices ( flashed with tasmota) that I use for my kitchen cabinet lighting. After approx 18 months of use and up until a few days ago they have worked flawlessly using alexa to control brightness and on/off functions. All of a sudden they no longer work with alexa. Only the first friendly name is detected, if you delete this within the alexa app, then the next friendly name is detected  and so on. Switching the mode to belkin wemo will detect all the friendly names and will switch outputs on and off but with no dimming function.

**Reply by CancunManny** [reply]
Hell TvTim, I don't have any real answers, but I am trying to collect as much info as possible.  I was under the impression this affected "new" devices.  In your case you had Alexa discover the devices about 18 months ago and worked fine, did it all of the sudden stop working?  Or did you try to add new devices, found it would only detect one, and while trying to figure out what was going on you deleted all the devices it had previously discovered and tried to discover them again?

If it stopped working without you making any other changes I think there would be a lot more people popping up here talking about the issue.

**Comment by rebel7580** [comment]
In the last few days, I've had not only several Tasmota "hue" devices
dropped by Alexa, but also all of my other devices that I exposed to Alexa
via a custom "hue emulator" (not related to Tasmota at all).
So I'm beginning to suspect that Amazon changed something in its HUE
interface that is causing our stuff to break.

**Reply by CancunManny** [reply]
thanks for great feedback rebel, seems things are getting worse.  You mentioned " but also all of my other devices that I exposed to Alexa
via a custom "hue emulator" (not related to Tasmota at all)."  Are those devices you mentioned actual hue lights?

I am planning on reaching out to Amazon via email, hoping they can escalate it to higher support tier.  Not sure if they are trying to close the hue emulator loophole (basically make it so that only real hue lights are detected) or if it is affecting even real hue lights.  Someone mentioned this on a this thread "However, I believe that Amazon has changed something in the HUE recognition that is causing this problem. Only multi-channel devices that are addressed directly via Alexa are likely to be affected. Anyone working with the newer HUE hubs (without Tasmota) also needs an additional app, if I'm correctly informed."

**Reply by rebel7580** [reply]
No, not "real" Hue lights. Hence the need for the Hue emulator ...

**Comment by s-hadinger** [comment]
The good news is that I can reproduce the problem. The bad news is that the confusion in devices happens at Alexa side. I will try to find a way to circumvent it

**Comment by CancunManny** [comment]
Good news guys!  Seems soon we will have a fix.  Issue had to do with way tasmota hue emulator was creating the IDs

> "uniqueid": "78:e3:6d:09:1d:a4:00:11-01"
> "uniqueid": "78:e3:6d:09:1d:a4:00:11-02"
> "uniqueid": "78:e3:6d:09:1d:a4:00:11-03"

Seems Alexa started to ignore everything past the - so all found devices were getting same ID.  Will post how to fix once the fix has been finalized!

**Reply by s-hadinger** [reply]
I should have a fix soon. I'm not sure I'll have time tomorrow, max next day

**Comment by s-hadinger** [comment, marked as answer]
This should be fixed now with the latest build from development

**Reply by TvTim** [reply]
Thanks, I will try it at the weekend.

**Reply by CancunManny** [reply]
I did a quick test on one of my tasmotas, and it worked as expected!   Now just have to wait until I have some free time to update the rest, a some of them require me doing custom builds!  Overall I am very happy Hue Emulator in Tasmota continues to work!

**Reply by TheLukeDude** [reply]
_On my side it did not work at all unfortunately. Is there a way I can provide CLI logs from my devices or so to help fix the issue?_
Edit>> removed all devices in Alexa app, searched for them again, now it works as expected

**Reply by CancunManny** [reply]
I am not very good with logs...  First you did install the last build of tasmota?  If so, is Alexa finding one device or no devices?

**Reply by TheLukeDude** [reply]
Ah, maybe that’s where the error is. Do I need to re-pair all devices with Alexa? I installed the latest dev build but did not delete devices in Alexa’s device list

**Reply by s-hadinger** [reply]
Yes you need to re-pair all devices after the patch - unless you had only one device

**Comment by jkwim** [comment]
Gosh! Was scratching my head unable to figure out why Alexa was mis-behaving!
Now I need to redo the custom compilations :-)
